### PR TITLE
doc: nrf: Add nRF700X driver entry

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

File: doc/nrf/drivers/nrf700x.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+***************
+The nRF700X Wi-Fi driver supports radio test mode, which can be used to test the RF performance of the nRF700X device. This is a build time option, and can be enabled by
+:kconfig:option:`CONFIG_NRF700X_RADIO_TEST` Kconfig option.
+
+The radio test mode is implemented as a separate application, and the application is located in :file:`samples/wifi/radio_test/` folder.
+
+API documentation
+*****************
+
+| Source folder: :file:`drivers/wifi/nrf700x/`
```
Comment (richabp) [inline doc/nrf/drivers/nrf700x.rst:66]: Do we have any header file and source file to mention?

**Review by richabp — COMMENTED** [review]
Image needs rework as per the style guide.

File: doc/nrf/drivers/nrf700x.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+nRF700X Wi-Fi Driver
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This driver implements the Wi-Fi protocol for the nRF700X FullMAC series of devices. FullMAC devices implement the Wi-Fi
+protocol in the chipset, and the driver only needs to configure the chipset and transfer the frames to and from the device
+to networking stack.
```
[the remark this answers is not in the document]
Comment (kittydepa) [reply doc/nrf/drivers/nrf700x.rst:12]: * Edited @richabp's suggestion for one additional improvement

File: doc/nrf/drivers/nrf700x.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+nRF700X is a companion chip and can be used with any SoC, primarily with Nordic Semiconductor's nRF53 and nRF91 Series SoCs.
```
[the remark this answers is not in the document]
Comment (kittydepa) [reply doc/nrf/drivers/nrf700x.rst:14]: Edited @richabp's suggestion for one additional improvement

File: doc/nrf/drivers/nrf700x.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+Driver can be enabled by using the :kconfig:option:`CONFIG_WIFI_NRF700X` Kconfig option.
```
[the remark this answers is not in the document]
Comment (kittydepa) [reply doc/nrf/drivers/nrf700x.rst:16]: Edited @richabp's suggestion for one additional improvement

File: doc/nrf/drivers/nrf700x.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+.. figure:: /images/nrf700x_wifi_driver.png
+   :alt: nRF700X Wi-Fi driver block diagram
+   :align: center
+   :figclass: align-center
+
+   nRF700X Wi-Fi driver block diagram
+
+Design overview
+***************
+The nRF700X Wi-Fi driver follows OS agnostic design, and the driver is implemented in to OS agnostic and OS (Zephyr) specific implementation files.
```
[the remark this answers is not in the document]
Comment (kittydepa) [reply doc/nrf/drivers/nrf700x.rst:32]: Edited @richabp's suggestion for one additional improvement

File: doc/nrf/drivers/nrf700x.rst [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+* Wi-Fi Direct mode - not supported yet
+
+Except for scan  only mode, the driver uses `hostapd` to implement AP MLME and `wpa_supplicant` to implement 802.1X supplicant.
+
+Driver to nRF700X communication
+*******************************
+The driver communicates with the nRF700X device using the QSPI/SPI interface. The driver uses the QSPI/SPI interface to send commands to the nRF700X device, and to transfer the
+data to and from the device. nRF7002 DK uses QSPI where as nRF7002 EK uses SPI.
+
+For nRF7002 EK, a new shield `nrf7002_ek` is required to connect the nRF7002 EK to the SoC.
```
[the remark this answers is not in the document]
Comment (kittydepa) [reply doc/nrf/drivers/nrf700x.rst:53]: Edited @richabp's suggestion for one additional improvement

File: doc/nrf/drivers/nrf700x.rst [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+The radio test mode is implemented as a separate application, and the application is located in :file:`samples/wifi/radio_test/` folder.
+
+API documentation
+*****************
+
+| Source folder: :file:`drivers/wifi/nrf700x/`
+
+After the nrf700x driver has been initialized, the application will see it as an Ethernet interface.
+To use that, the application can use `Zephyr Network APIs`_.
```
Comment (kittydepa) [inline doc/nrf/drivers/nrf700x.rst:69]:
```suggestion
To use the Ethernet interface, the application can use `Zephyr Network APIs`_.
```

**Review by kittydepa — CHANGES_REQUESTED** [review]

File: doc/nrf/drivers/nrf700x.rst [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+The driver is designed to be used with the Zephyr networking stack, and the driver is implemented as a network interface driver. Driver implements various IEEE082.11 Features
+and supports the following features:
+
+* Wi-Fi 6 (802.11ax) support
+* WPA3/WPA2 personal security
+* IEEE 802.11 power save modes
+* Scan only mode
+* IEEE 802.11 Station (STA) mode
+* IEEE 802.11 SoftAP mode - not supported yet
+* Wi-Fi Direct mode - not supported yet
```
[the remark this answers is not in the document]
Comment (krish2718) [reply doc/nrf/drivers/nrf700x.rst:47]: Well, these are part of the driver code already, just that full end-end support is not added yet.

File: doc/nrf/drivers/nrf700x.rst (hunk 1 above)
Comment (krish2718) [reply doc/nrf/drivers/nrf700x.rst:72]: there are too many, so, just pointed to the folder

File: doc/nrf/drivers/nrf700x.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (kittydepa) [reply doc/nrf/drivers/nrf700x.rst:47]: Perhaps the last two can be added to its own list stating:

> The following features are in the driver code but not yet supported:
> * IEEE 802.11 SoftAP mode
> * Wi-Fi Direct mode

File: doc/nrf/drivers/nrf700x.rst [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+   :alt: nRF700X Wi-Fi driver block diagram
+   :align: center
+   :figclass: align-center
+
+   nRF700X Wi-Fi driver architecture overview
+
+Design overview
+***************
+
+The nRF700X Wi-Fi driver follows a OS agnostic design, and the driver is implemented into OS agnostic and OS (Zephyr) specific implementation files.
```
Comment (kittydepa) [inline doc/nrf/drivers/nrf700x.rst:33]:
```suggestion
The nRF700X Wi-Fi driver follows an OS agnostic design, and the driver is implemented into OS agnostic and OS (Zephyr) specific implementation files.
```

**Review by kittydepa — CHANGES_REQUESTED** [review]
One small suggestion that I missed earlier. Otherwise LGTM

**Review by kittydepa — APPROVED** [review]

File: doc/nrf/drivers/nrf700x.rst [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+*******************************
+
+The driver communicates with the nRF700X device using the QSPI/SPI interface.
+The driver uses the QSPI/SPI interface to send commands to the nRF700X device, and to transfer the data to and from the device.
+The nRF7002 DK uses QSPI whereas the nRF7002 EK uses SPI.
+
+To connect the nRF7002 EK to the SoC, the ``nrf7002_ek`` shield is required.
+
+
+Radio test mode
```
Comment (richabp) [inline doc/nrf/drivers/nrf700x.rst:64]:
```suggestion
Radio Test mode
```

File: doc/nrf/drivers/nrf700x.rst [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+***************
+
+The nRF700X Wi-Fi driver supports radio test mode, which can be used to test the RF performance of the nRF700X device.
```
Comment (richabp) [inline doc/nrf/drivers/nrf700x.rst:67]:
```suggestion
The nRF700X Wi-Fi driver supports Radio Test mode, which can be used to test the RF performance of the nRF700X device.
```

File: doc/nrf/drivers/nrf700x.rst [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+This is a build time option and can be enabled by using the :kconfig:option:`CONFIG_NRF700X_RADIO_TEST` Kconfig option.
+
+For more details about using this driver in radio test mode, see :ref:`wifi_radio_test`.
```
Comment (richabp) [inline doc/nrf/drivers/nrf700x.rst:70]:
```suggestion
For more details about using this driver in Radio Test mode, see :ref:`wifi_radio_test`.
```

**Review by richabp — CHANGES_REQUESTED** [review]
Please add in the Drivers section of the Changelog:

* Added :ref:`nrf700x_wifi`.

**Review by richabp — APPROVED** [review]

**Pull request merged**